benchadapt/python/benchadapt/run.py

```diff
@@ -133,15 +136,7 @@ def to_publishable_dict(self):
                 "Run not publishable! `machine_info` xor `cluster_info` must be specified"
             )
 
-        if not (
-            res_dict["github"].get("repository") and res_dict["github"].get("commit")
-        ):
-            raise ValueError(
-                "Run not publishable! `github.repository` and `github.commit` must be populated. "
-                "You may pass github metadata via CONBENCH_PROJECT_REPOSITORY, CONBENCH_PROJECT_COMMIT, "
-                "and CONBENCH_PR_NUMBER environment variables. "
-                f"\ngithub: {res_dict['github']}"
-            )
+        validate_or_remove_github_commit_key(res_dict, strict=True)
```

Comment:
We had duplicate code here, I thought it's better to de-duplicate, especially now that I added logic.

Of course there were minor differences between the two code sections, they are now abstracted with the `strict` parameter.